### ISSUE TITLE
Can't install on RockyLinux

### ISSUE THREAD

**Issue by fnordfish** [kgiszczak/tomlib#4]
While `gem install tomlib` seems to finish successful, `require "tomlib"` cashes with an `LoadError`.

Tested in Docker Images for rockylinux 8 and 9.

```Dockerfile
FROM rocklinux:9
RUN dnf upgrade -y && \
dnf module enable ruby:3.3 -y && \
dnf groupinstall "Development Tools" && \
dnf install -y --enablerepo=devel \
    ruby-3.3.5 \
    ruby-devel \
    ruby-bundled-gems \
    ruby-default-gems \
    rubygem-irb \
    rubygem-rake \
    openssl-devel \
    libyaml-devel \
    zlib-devel \
    libffi-devel
```

## Installing

The generated Makefile seems to have a missing (`false`) target. This seems odd: `rm -fr tomlib.so false toml.o tomlib.o *.bak mkmf.log .*.time`

Full install log:

```console
# gem install --verbose tomlib
HEAD https://index.rubygems.org/
200 OK
GET https://index.rubygems.org/info/tomlib
200 OK
GET https://index.rubygems.org/info/bigdecimal
200 OK
Downloading gem tomlib-0.7.2.gem
GET https://index.rubygems.org/gems/tomlib-0.7.2.gem
Fetching tomlib-0.7.2.gem
200 OK
/usr/local/share/gems/gems/tomlib-0.7.2/CHANGELOG.md
/usr/local/share/gems/gems/tomlib-0.7.2/LICENSE.txt
/usr/local/share/gems/gems/tomlib-0.7.2/README.md
/usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib/extconf.rb
/usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib/toml.c
/usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib/toml.h
/usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib/tomlib.c
/usr/local/share/gems/gems/tomlib-0.7.2/lib/tomlib.rb
/usr/local/share/gems/gems/tomlib-0.7.2/lib/tomlib/dumper.rb
/usr/local/share/gems/gems/tomlib-0.7.2/lib/tomlib/error.rb
/usr/local/share/gems/gems/tomlib-0.7.2/lib/tomlib/version.rb
/usr/local/share/gems/gems/tomlib-0.7.2/tomlib.gemspec
Building native extensions. This could take a while...
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
["/usr/bin/ruby", "-I/usr/share/rubygems", "extconf.rb"]
creating Makefile
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-1504-guexnh", "sitelibdir=./.gem.20241129-1504-guexnh", "clean"]
rm -f
rm -fr tomlib.so false toml.o tomlib.o *.bak mkmf.log .*.time
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-1504-guexnh", "sitelibdir=./.gem.20241129-1504-guexnh"]
gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.   -fPIC -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -fasynchronous-unwind-tables -fstack-clash-protection -fPIC  -o toml.o -c toml.c
gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I.   -fPIC -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -fasynchronous-unwind-tables -fstack-clash-protection -fPIC  -o tomlib.o -c tomlib.c
rm -f tomlib.so
gcc -shared -o tomlib.so toml.o tomlib.o -L. -L/usr/lib64 -L. -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -fstack-protector-strong -rdynamic -Wl,-export-dynamic -Wl,--no-as-needed -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1     -lruby  -lm -lpthread  -lc
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-1504-guexnh", "sitelibdir=./.gem.20241129-1504-guexnh", "install"]
/usr/bin/mkdir -p . ./.gem.20241129-1504-guexnh/tomlib
exit > .sitearchdir.-.tomlib.time
/usr/bin/install -c -m 0755 tomlib.so ./.gem.20241129-1504-guexnh/tomlib
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-1504-guexnh", "sitelibdir=./.gem.20241129-1504-guexnh", "clean"]
rm -f
rm -fr tomlib.so false toml.o tomlib.o *.bak mkmf.log .*.time
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
/usr/bin/ruby -I/usr/share/rubygems extconf.rb
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-1504-guexnh sitelibdir\=./.gem.20241129-1504-guexnh clean
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-1504-guexnh sitelibdir\=./.gem.20241129-1504-guexnh
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-1504-guexnh sitelibdir\=./.gem.20241129-1504-guexnh install
current directory: /usr/local/share/gems/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-1504-guexnh sitelibdir\=./.gem.20241129-1504-guexnh clean
Successfully installed tomlib-0.7.2
Parsing documentation for tomlib-0.7.2
Parsing sources...
100% [ 9/ 9]  lib/tomlib/version.rb
Installing ri documentation for tomlib-0.7.2
Done installing documentation for tomlib after 0 seconds
1 gem installed
```

## Loading

```console
# irb -r tomlib
/usr/share/gems/gems/irb-1.13.1/lib/irb/init.rb:436: warning: LoadError: cannot load such file -- /usr/local/share/gems/gems/tomlib-0.7.2/lib/tomlib/tomlib
```

## Investigating

```ruby
$LOAD_PATH.grep /tomlib/
# => ["/usr/local/share/gems/gems/tomlib-0.7.2/lib", "/usr/local/lib64/gems/ruby/tomlib-0.7.2"]
```

There is a `/usr/local/lib64/gems/ruby/tomlib-0.7.2/tomlib/tomlib.so`, but 

```console
# ls -la /usr/local/share/gems/gems/tomlib-0.7.2/lib/tomlib
total 24
drwxr-xr-x 2 root root 4096 Nov 29 11:24 .
drwxr-xr-x 3 root root 4096 Nov 29 11:24 ..
-rw-r--r-- 1 root root 4831 Nov 29 11:24 dumper.rb
-rw-r--r-- 1 root root  153 Nov 29 11:24 error.rb
-rw-r--r-- 1 root root   86 Nov 29 11:24 version.rb
```

When I manually copy the `.so` into the gem dir, everything works.

In contrast, on a `ruby:buster` we get:

```console
# gem install --verbose tomlib
HEAD https://index.rubygems.org/
200 OK
GET https://index.rubygems.org/info/tomlib
200 OK
GET https://index.rubygems.org/info/bigdecimal
200 OK
Downloading gem tomlib-0.7.2.gem
GET https://index.rubygems.org/gems/tomlib-0.7.2.gem
Fetching tomlib-0.7.2.gem
200 OK
/usr/local/bundle/gems/tomlib-0.7.2/CHANGELOG.md
/usr/local/bundle/gems/tomlib-0.7.2/LICENSE.txt
/usr/local/bundle/gems/tomlib-0.7.2/README.md
/usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib/extconf.rb
/usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib/toml.c
/usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib/toml.h
/usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib/tomlib.c
/usr/local/bundle/gems/tomlib-0.7.2/lib/tomlib.rb
/usr/local/bundle/gems/tomlib-0.7.2/lib/tomlib/dumper.rb
/usr/local/bundle/gems/tomlib-0.7.2/lib/tomlib/error.rb
/usr/local/bundle/gems/tomlib-0.7.2/lib/tomlib/version.rb
/usr/local/bundle/gems/tomlib-0.7.2/tomlib.gemspec
Building native extensions. This could take a while...
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
["/usr/local/bin/ruby", "extconf.rb"]
creating Makefile
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-46-td30nk", "sitelibdir=./.gem.20241129-46-td30nk", "clean"]

current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-46-td30nk", "sitelibdir=./.gem.20241129-46-td30nk"]
compiling toml.c
compiling tomlib.c
linking shared-object tomlib/tomlib.so
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-46-td30nk", "sitelibdir=./.gem.20241129-46-td30nk", "install"]
/usr/bin/install -c -m 0755 tomlib.so ./.gem.20241129-46-td30nk/tomlib
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
["make", "DESTDIR=", "sitearchdir=./.gem.20241129-46-td30nk", "sitelibdir=./.gem.20241129-46-td30nk", "clean"]

current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
/usr/local/bin/ruby extconf.rb
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-46-td30nk sitelibdir\=./.gem.20241129-46-td30nk clean
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-46-td30nk sitelibdir\=./.gem.20241129-46-td30nk
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-46-td30nk sitelibdir\=./.gem.20241129-46-td30nk install
current directory: /usr/local/bundle/gems/tomlib-0.7.2/ext/tomlib
make DESTDIR\= sitearchdir\=./.gem.20241129-46-td30nk sitelibdir\=./.gem.20241129-46-td30nk clean
Successfully installed tomlib-0.7.2
1 gem installed
```

```
/usr/local/bundle/gems/tomlib-0.7.2/lib/tomlib/tomlib.so
/usr/local/bundle/extensions/aarch64-linux/3.2.0/tomlib-0.7.2/tomlib/tomlib.so
```

**Comment by fnordfish** [comment]
Found this quite old comment (https://bugs.ruby-lang.org/issues/9106#note-8), that helped me realize the issue.
See Fix in #5

**Comment by kgiszczak** [comment]
Fix is merged
